srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/PopulatedDisplayGroup.java

```diff
@@ -0,0 +1,204 @@
+ */
+public final class PopulatedDisplayGroup implements Comparable<PopulatedDisplayGroup>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(PopulatedDisplayGroup.class);
+    
+    private final List<DisplayItem> fDisplayItems;
+    
+    private final List<Variable> fVariables;
+    
+    /**
+     * Constructs an instance from the given List of variables. Also populates the groups with
+     * any patient notes that are automatically retrieved.
+     * @param variables must have at least one member
+     * @throws IllegalArgumentException if the given list is empty or if it
+     * contains Variables with different groups
+     */
+    public PopulatedDisplayGroup(final List<? extends Variable> variables, final Patient patient)
+    {
+        // Check precondition: variable list is not empty
+        if (variables.isEmpty())
+        {
+            throw new IllegalArgumentException("variable list must not be empty");
+        }
+        
+        // Check precondition: variables all have the same group
+        final VariableGroup group = variables.get(0).getGroup();
+        for (final Variable var : variables)
+        {
+            if (!var.getGroup().equals(group))
+            {
+                throw new IllegalArgumentException(
+                        "variables do not all have the same group");
+            }
+        }
+        
+        // Preconditions satisfied: build views of all of the necessary variables
+        // so that they can be represented as DisplayItem instances.
+        // Store variables for reference later.
+        fVariables = new ArrayList<>(variables);
+        fDisplayItems = new ArrayList<>();
+        for(final Variable var: variables)
+        {
+            // TODO: If the variable has specified keys that have reference information,
+            // create a String that holds that reference information and use it in the view constructors.
+            VariableView varView;
+            if(var instanceof BooleanVariable)
+            {
+                varView = new BooleanVariableView((BooleanVariable) var, "");
+            }
+            else if(var instanceof DiscreteNumericalVariable)
+            {
+                varView = new DiscreteNumericalVariableView((DiscreteNumericalVariable) var, "");
+            }
+            else if(var instanceof NumericalVariable)
+            {
+                varView = new NumericalVariableView((NumericalVariable) var, "");
+            }
+            else if(var instanceof MultiSelectVariable)
+            {
+                varView = new MultiSelectVariableView((MultiSelectVariable) var, "");
+            }
+            else if(var instanceof ProcedureVariable)
+            {
+                varView = new ProcedureVariableView((ProcedureVariable) var, "");
+            }
+            else
+            {
+                LOGGER.debug("Unexpected Variable type while creating a PopulatedDisplayGroup. Key: {}",
```

Comment:
This is a DEBUG-level log?? I think it should be an exception: it's a coding error.